### DISCUSSION TITLE
division without rimander

### DISCUSSION THREAD

**Discussion by Drugo67** [VBAndCs/sVB-Small-Visual-Basic#92] [Q&A]
Hello again. I would need to do a division and keep only the integer without decimals. For example 12 / 13 = 0.923 but I only need to keep the 0. If I use math.round the division is approximated to 1 and the same goes with math.ceiling. in vb6 I did it with k = int(0.9) and the result was 0. In sVB I had to transform it into a string (text.tostr) and take the integer part with an "if then" checking character by character up to the decimal point and transform it back into an integer (text.tonumber). Isn't there something easier like it was in vb6?

**Comment by VBAndCs** [comment]
I didn't add the Truncate function, but it is easy to write:
TW.WriteLine(Truncate(0.9))
TW.WriteLine(Truncate(-0.9))

' ------------------------------------------------
Function Truncate(n)
   If n < 0 Then
      Return Math.Ceiling(n)
   Else
      Return Math.Floor(n)
   EndIf
EndFunction

Please don't forget to close this issue.
Thanks.
________________________________
From: Drugo67 ***@***.***>
Sent: Sunday, October 20, 2024 3:55 PM
To: VBAndCs/sVB-Small-Visual-Basic ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [VBAndCs/sVB-Small-Visual-Basic] division without rimander (Discussion #92)


Hello again. I would need to do a division and keep only the integer without decimals. For example 12 / 13 = 0.923 but I only need to keep the 0. If I use math.round the division is approximated to 1 and the same goes with math.ceiling. in vb6 I did it with k = int(0.9) and the result was 0. In sVB I had to transform it into a string (text.tostr) and take the integer part with an "if then" checking character by character up to the decimal point and transform it back into an integer (text.tonumber). Isn't there something easier like it was in vb6?

**Reply by Drugo67** [reply]
Thank You Mohammad (as always). Ok. So, if I only want the integer without decimal (and without rounding) I have to use math.floor for the positives and math.ceiling for the negatives. Thank You so much

**Comment by VBAndCs** [comment]
I also added the Math.Truncate method, and it will be available in the next release.
________________________________
